net/net.c

```diff
@@ -196,18 +196,26 @@ M0_EXPORTED(m0_net_xprt_register);
 M0_INTERNAL void m0_net_xprt_deregister(const struct m0_net_xprt *xprt)
 {
 	int i;
-
+	int j;
 	for (i = 0; i < ARRAY_SIZE(xprts); ++i) {
 		if (xprts[i] == xprt) {
-			xprts[i] = NULL;
 			if (xprt == xprt_default)
 				xprt_default = NULL;
+			for (j = i; j < ARRAY_SIZE(xprts) - 1; ++j)
+				xprts[j] = xprts[j + 1];
 			return;
 		}
 	}
 	M0_IMPOSSIBLE("Wrong xprt.");
 }
 M0_EXPORTED(m0_net_xprt_deregister);
+
+void print_xprt(void)
+{
+	int i;
+	for (i = 0; i < ARRAY_SIZE(xprts) -1; ++i)
```

Comment:
Why is the upper limit "ARRAY_SIZE(xprts) -1"?
Should it be:
```
for (i = 0; i < ARRAY_SIZE(xprts); ++i)
```

And another bug:
"xprts[i]" might be NULL.
It should be check before dereferencing its ->nx_name.

Reply:
incorporated